src/model.ts

```diff
@@ -523,23 +530,29 @@ export class Model extends EventBus<any> implements CommandDispatcher {
         }
         this.status = Status.Running;
         const { changes, commands } = this.state.recordChanges(() => {
+          const start = performance.now();
+          console.group(`Dispatching of ${type}`);
           if (isCoreCommand(command)) {
             this.state.addCommand(command);
           }
           this.dispatchToHandlers(this.handlers, command);
           this.finalize();
+          console.groupEnd();
+          console.info(`Command ${type} dispatched in`, performance.now() - start, "ms");
         });
         this.session.save(command, commands, changes);
         this.status = Status.Ready;
         this.trigger("update");
         break;
       case Status.Running:
+        const start = performance.now();
         if (isCoreCommand(command)) {
           const dispatchResult = this.checkDispatchAllowed(command);
           if (!dispatchResult.isSuccessful) {
             return dispatchResult;
           }
           this.state.addCommand(command);
+          console.debug(type, performance.now() - start, "ms");
```

Comment:
I would not do that. It logs *every* sub-command dispatched by UI plugins, which might be a looooot (autofilling 10k rows).
Logging is probably not free performance-wise.